Modules/_ctypes/callbacks.c

```diff
@@ -308,7 +313,12 @@ static void _CallPythonObject(void *mem,
     Py_XDECREF(result);
 
   Done:
-    Py_XDECREF(arglist);
+    for (Py_ssize_t i = 0; i < nArgs; i++) {
+        Py_XDECREF(args[i]);
```

Comment:
You access uninitiallized memory if `for (Py_ssize_t i = 0; i < nArgs; ++i) {` loop call `goto Done`. Maybe use another variable to store the number of initialized items, or update nArgs on error, or use PyMem_Calloc() (slower?), etc.